engine/runtime/src/main/scala/org/enso/compiler/codegen/IrToTruffle.scala

```diff
@@ -375,7 +377,17 @@ class IrToTruffle(
                 ) Right(None)
                 else Left(l)
               )
-              .map(_.filterNot(_ => cons.isBuiltin))
+              .map(fOpt =>
+                // Register builtin iff
+                // - a method refers to a builtin with a diferent name or owner
```

Comment:
Could you elaborate what this first case means?

Is it for situations where we have
```
type Foo
    baz = @Builtin_Method "Bar.quux"
```
or something similar?
Or something else?

Reply:
Yeah. For now `Platform.system` is the prime example.

Reply:
Do you think that we should allow that at all? IMO it's better to move the stub to the right place so it is kept in sync. Otherwise sounds like a way to create a spaghetti mess.

Reply:
I would tend to agree that this might lead to some confusion that's why I would suggest to use it very carefully.
But like I said, there was at least one occasion already when this was requested (even by you I believe ;) ).
And I made typo - it wasn't `Platform.system` but rather `Platform.is_unix` which is a stub for `System.is_unix` builtin.
If we remove this, you will have to make workarounds like with `Platform.os` which calls `System.os`.

Reply:
Hmmmm, I see, well tbh I don't remember requesting it _this way_. At least right now my opinion is that it's better to have `Platform.os` delegate to `System.os` if needed, but have the builtins placed in sensible places instead of having the ability to connect them completely randomly.

Reply:
After checking again, looks like this case won't be needed as the new condition handles that as well. I will remove that.